client/landing/stepper/declarative-flow/new-hosted-site-flow.ts

```diff
@@ -22,11 +22,26 @@ import type { OnboardSelect, UserSelect } from '@automattic/data-stores';
 import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
 import './internals/new-hosted-site-flow.scss';
 
+function useShowDomainStep(): boolean {
+	const query = useQuery();
+	const showDomainStepValue = query.get( 'showDomainStep' );
+	return showDomainStepValue !== null && showDomainStepValue.toLowerCase() !== 'false';
+}
+
 const hosting: Flow = {
 	name: NEW_HOSTED_SITE_FLOW,
 	isSignupFlow: true,
 	useSteps() {
+		const showDomainStep = useShowDomainStep();
```

Comment:
I find it awkward setting a const that's just a bool. Can we use `useShowDomainStep()` directly in the condition?

Reply:
Of course we can, I have asked `Cursor` and he agrees with the current version 
![CleanShot 2024-10-29 at 17 45 02@2x](https://github.com/user-attachments/assets/0526a82c-e833-4386-be16-b283b9609f96)

Reply:
I had literally just installed Cursor 😁